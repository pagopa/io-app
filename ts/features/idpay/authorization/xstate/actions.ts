import {
  AppParamsList,
  IOStackNavigationProp
} from "../../../../navigation/params/AppParamsList";
import { IDPayAuthorizationRoutes } from "../navigation/navigator";

const createActionsImplementation = (
  navigation: IOStackNavigationProp<AppParamsList, keyof AppParamsList>
) => {
  const navigateToConfirmationScreen = () => {
    navigation.navigate(IDPayAuthorizationRoutes.IDPAY_AUTHORIZATION_MAIN, {
      screen: IDPayAuthorizationRoutes.IDPAY_AUTHORIZATION_CONFIRM
    });
  };

  const navigateToResultScreen = () =>
    navigation.navigate(IDPayAuthorizationRoutes.IDPAY_AUTHORIZATION_MAIN, {
      screen: IDPayAuthorizationRoutes.IDPAY_AUTHORIZATION_RESULT
    });

  const exitAuthorization = () => {
    navigation.pop();
  };

  return {
    navigateToConfirmationScreen,
    navigateToResultScreen,
    exitAuthorization
  };
};

export { createActionsImplementation };
