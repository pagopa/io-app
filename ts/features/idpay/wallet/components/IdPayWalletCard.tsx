import React from "react";
import { Pressable } from "react-native";
import { useIONavigation } from "../../../../navigation/params/AppParamsList";
import { withWalletCardBaseComponent } from "../../../newWallet/components/WalletCardBaseComponent";
import { IDPayDetailsRoutes } from "../../details/navigation";
import { IdPayCard, IdPayCardProps } from "./IdPayCard";

export type IdPayWalletCardProps = IdPayCardProps & {
  initiativeId: string;
};

const WrappedIdPayCard = (props: IdPayWalletCardProps) => {
  const navigation = useIONavigation();

  const { initiativeId, ...cardProps } = props;

  const handleOnPress = () => {
    navigation.navigate(IDPayDetailsRoutes.IDPAY_DETAILS_MAIN, {
      screen: IDPayDetailsRoutes.IDPAY_DETAILS_MONITORING,
      params: {
        initiativeId
      }
    });
  };

  return (
    <Pressable accessibilityRole="button" onPress={handleOnPress}>
      <IdPayCard {...cardProps} />
    </Pressable>
  );
};

/**
 * Wrapper component which adds wallet capabilites to the IdPayCard component
 */
export const IdPayWalletCard = withWalletCardBaseComponent(WrappedIdPayCard);
