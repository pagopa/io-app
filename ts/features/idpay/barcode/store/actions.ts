import { ActionType, createAsyncAction } from "typesafe-actions";
import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
import { TransactionErrorDTO } from "../../../../../definitions/idpay/TransactionErrorDTO";
import { NetworkError } from "../../../../utils/errors";

export type IdPayGenerateBarcodePayload = {
  initiativeId: string;
};

export const idPayGenerateBarcode = createAsyncAction(
  "IDPAY_GENERATE_BARCODE_REQUEST",
  "IDPAY_GENERATE_BARCODE_SUCCESS",
  "IDPAY_GENERATE_BARCODE_FAILURE"
)<
  IdPayGenerateBarcodePayload,
  TransactionBarCodeResponse,
  { initiativeId: string; error: TransactionErrorDTO | NetworkError }
>();

export type IdPayBarcodeActions = ActionType<typeof idPayGenerateBarcode>;
