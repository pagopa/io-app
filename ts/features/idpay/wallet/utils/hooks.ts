import { useFocusEffect } from "@react-navigation/native";
import * as React from "react";
import { useSelector } from "react-redux";
import { useIODispatch } from "../../../../store/hooks";
import { idPayInitiativesFromInstrumentGet } from "../store/actions";
import {
  idPayAreInitiativesFromInstrumentErrorSelector,
  idPayEnabledInitiativesFromInstrumentSelector
} from "../store/reducers";

export const useIDPayInitiativesFromInstrument = (idWallet: string) => {
  const dispatch = useIODispatch();
  const namedInitiativesList = useSelector(
    idPayEnabledInitiativesFromInstrumentSelector
  );
  const areInitiativesInError = useSelector(
    idPayAreInitiativesFromInstrumentErrorSelector
  );
  React.useEffect(() => {
    // required to clear any old data
    dispatch(
      idPayInitiativesFromInstrumentGet.request({
        idWallet,
        isRefreshCall: false
      })
    );
  }, [idWallet, dispatch]);
  useFocusEffect(() => {
    // will be removed in #IODPAY-207 in favor of saga logic
    const timer = setInterval(
      () => {
        dispatch(
          idPayInitiativesFromInstrumentGet.request({
            idWallet,
            isRefreshCall: true
          })
        );
      },
      areInitiativesInError ? 6000 : 3000
    );
    return () => clearInterval(timer);
  });

  return {
    namedInitiativesList
  };
};
