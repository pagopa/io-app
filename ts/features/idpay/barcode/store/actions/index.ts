import { ActionType, createAsyncAction } from "typesafe-actions";
import { TransactionBarCodeResponse } from "../../../../../../definitions/idpay/TransactionBarCodeResponse";
import { TransactionErrorDTO } from "../../../../../../definitions/idpay/TransactionErrorDTO";
import { NetworkError } from "../../../../../utils/errors";

export type IdPayGenerateBarcodePayload = {
  initiativeId: string;
};

export const idPayGenerateBarcode = createAsyncAction(
  "IDPAY_GENERATE_BARCODE_REQUEST",
  "IDPAY_GENERATE_BARCODE_SUCCESS",
  "IDPAY_GENERATE_BARCODE_FAILURE"
)<
  IdPayGenerateBarcodePayload,
  TransactionBarCodeResponse,
  {
    initiativeId: string;
    error: TransactionErrorDTO | NetworkError;
  }
>();

export type IdPayBarcodeActions = ActionType<typeof idPayGenerateBarcode>;

export const idPayGenerateStaticCode = createAsyncAction(
  "IDPAY_GENERATE_STATIC_CODE_REQUEST",
  "IDPAY_GENERATE_STATIC_CODE_SUCCESS",
  "IDPAY_GENERATE_STATIC_CODE_FAILURE"
)<
  IdPayGenerateBarcodePayload,
  TransactionBarCodeResponse,
  {
    initiativeId: string;
    error: TransactionErrorDTO | NetworkError;
  }
>();

export type IdPayStaticCodeActions = ActionType<typeof idPayGenerateStaticCode>;
