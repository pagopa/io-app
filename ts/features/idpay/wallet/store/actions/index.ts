import {
  ActionType,
  createAsyncAction,
  createStandardAction
} from "typesafe-actions";
import { InitiativesWithInstrumentDTO } from "../../../../../../definitions/idpay/InitiativesWithInstrumentDTO";
import { WalletDTO } from "../../../../../../definitions/idpay/WalletDTO";
import { NetworkError } from "../../../../../utils/errors";

export type IdPayInitiativesFromInstrumentPayloadType = {
  idWallet: string;
  isRefreshing?: boolean;
};

export type IdpayInitiativesInstrumentEnrollPayloadType = {
  idWallet: string;
  initiativeId: string;
};

export type IdpayInitiativesInstrumentDeletePayloadType = {
  instrumentId: string;
  initiativeId: string;
};

type InitiativeId = { initiativeId: string };
type NetworkErrorWithInitiativeId = {
  error: NetworkError;
  initiativeId: string;
};

export const idPayWalletGet = createAsyncAction(
  "IDPAY_WALLET_REQUEST",
  "IDPAY_WALLET_SUCCESS",
  "IDPAY_WALLET_FAILURE"
)<void, WalletDTO, NetworkError>();

export const idPayInitiativesFromInstrumentGet = createAsyncAction(
  "IDPAY_INITIATIVES_FROM_INSTRUMENT_REQUEST",
  "IDPAY_INITIATIVES_FROM_INSTRUMENT_SUCCESS",
  "IDPAY_INITIATIVES_FROM_INSTRUMENT_FAILURE"
)<
  IdPayInitiativesFromInstrumentPayloadType,
  InitiativesWithInstrumentDTO,
  NetworkError
>();

export const idpayInitiativesInstrumentEnroll = createAsyncAction(
  "IDPAY_INITIATIVES_INSTRUMENT_ENROLL_REQUEST",
  "IDPAY_INITIATIVES_INSTRUMENT_ENROLL_SUCCESS",
  "IDPAY_INITIATIVES_INSTRUMENT_ENROLL_FAILURE"
)<
  IdpayInitiativesInstrumentEnrollPayloadType,
  InitiativeId,
  NetworkErrorWithInitiativeId
>();

export const idpayInitiativesInstrumentDelete = createAsyncAction(
  "IDPAY_INITIATIVES_INSTRUMENT_DELETE_REQUEST",
  "IDPAY_INITIATIVES_INSTRUMENT_DELETE_SUCCESS",
  "IDPAY_INITIATIVES_INSTRUMENT_DELETE_FAILURE"
)<
  IdpayInitiativesInstrumentDeletePayloadType,
  InitiativeId,
  NetworkErrorWithInitiativeId
>();

export const idPayInitiativesFromInstrumentRefreshStart = createStandardAction(
  "IDPAY_INITIATIVES_FROM_INSTRUMENT_REFRESH_START"
)<IdPayInitiativesFromInstrumentPayloadType>();

export const idPayInitiativesFromInstrumentRefreshStop = createStandardAction(
  "IDPAY_INITIATIVES_FROM_INSTRUMENT_REFRESH_STOP"
)();

export type IdPayWalletActions =
  | ActionType<typeof idPayWalletGet>
  | ActionType<typeof idPayInitiativesFromInstrumentGet>
  | ActionType<typeof idpayInitiativesInstrumentEnroll>
  | ActionType<typeof idpayInitiativesInstrumentDelete>
  | ActionType<typeof idPayInitiativesFromInstrumentRefreshStart>
  | ActionType<typeof idPayInitiativesFromInstrumentRefreshStop>;
