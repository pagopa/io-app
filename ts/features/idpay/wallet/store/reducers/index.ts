import * as pot from "@pagopa/ts-commons/lib/pot";
import { pipe } from "fp-ts/lib/function";
import { createSelector } from "reselect";
import { getType } from "typesafe-actions";
import { StatusEnum as InitiativeStatus } from "../../../../../../definitions/idpay/InitiativeDTO";
import { StatusEnum as InstrumentInitiativeStatus } from "../../../../../../definitions/idpay/InitiativesStatusDTO";
import { InitiativesWithInstrumentDTO } from "../../../../../../definitions/idpay/InitiativesWithInstrumentDTO";
import { WalletDTO } from "../../../../../../definitions/idpay/WalletDTO";
import { Action } from "../../../../../store/actions/types";
import { isIdPayEnabledSelector } from "../../../../../store/reducers/backendStatus";
import { GlobalState } from "../../../../../store/reducers/types";
import { NetworkError } from "../../../../../utils/errors";
import {
  idPayInitiativesFromInstrumentGet,
  idPayWalletGet,
  idpayInitiativesInstrumentDelete,
  idpayInitiativesInstrumentEnroll
} from "../actions";

export type IdPayWalletState = {
  initiatives: pot.Pot<WalletDTO, NetworkError>;
  initiativesWithInstrument: pot.Pot<
    InitiativesWithInstrumentDTO,
    NetworkError
  >;
  initiativesAwaitingStatusUpdate: Record<string, boolean>;
  // structure: {initiativeId: is waiting for response to pair/unpair api call}
  // this will be populated on selection and reset when not loading and
  // we have a response from BE
};

const INITIAL_STATE: IdPayWalletState = {
  initiatives: pot.none,
  initiativesWithInstrument: pot.none,
  initiativesAwaitingStatusUpdate: {}
};

const reducer = (
  state: IdPayWalletState = INITIAL_STATE,
  action: Action
): IdPayWalletState => {
  switch (action.type) {
    case getType(idPayWalletGet.request):
      return { ...state, initiatives: pot.toLoading(state.initiatives) };
    case getType(idPayWalletGet.success):
      return { ...state, initiatives: pot.some(action.payload) };
    case getType(idPayWalletGet.failure):
      return {
        ...state,
        initiatives: pot.toError(state.initiatives, action.payload)
      };
    // Initiatives with instrument
    case getType(idPayInitiativesFromInstrumentGet.request):
      if (!action.payload.isRefreshing) {
        return {
          ...state,
          initiativesWithInstrument: pot.noneLoading,
          initiativesAwaitingStatusUpdate: {}
        };
      }

      if (pot.isSome(state.initiativesWithInstrument)) {
        return {
          ...state,
          initiativesWithInstrument: pot.toUpdating(
            state.initiativesWithInstrument,
            state.initiativesWithInstrument.value
          )
        };
      }

      if (!pot.isError(state.initiativesWithInstrument)) {
        return {
          ...state,
          initiativesWithInstrument: pot.toLoading(
            state.initiativesWithInstrument
          )
        };
      }
      return {
        ...state,
        initiativesWithInstrument: state.initiativesWithInstrument
      };
    case getType(idPayInitiativesFromInstrumentGet.success):
      const initiativesToKeepInLoadingState = pipe(
        state.initiativesAwaitingStatusUpdate,
        Object.entries,
        // remove all entries that have completed their request
        entries => entries.filter(([_, value]) => value),
        Object.fromEntries
      );

      return {
        ...state,
        initiativesWithInstrument: pot.some(action.payload),
        initiativesAwaitingStatusUpdate: initiativesToKeepInLoadingState
      };
    case getType(idPayInitiativesFromInstrumentGet.failure):
      return {
        ...state,
        initiativesWithInstrument: pot.toError(
          state.initiativesWithInstrument,
          action.payload
        )
      };
    // initiative pairing
    case getType(idpayInitiativesInstrumentDelete.request):
    case getType(idpayInitiativesInstrumentEnroll.request):
      return {
        ...state,
        initiativesAwaitingStatusUpdate: {
          ...state.initiativesAwaitingStatusUpdate,
          [action.payload.initiativeId]: true
        }
      };
    case getType(idpayInitiativesInstrumentDelete.success):
    case getType(idpayInitiativesInstrumentEnroll.success):
    case getType(idpayInitiativesInstrumentDelete.failure):
    case getType(idpayInitiativesInstrumentEnroll.failure):
      return {
        ...state,
        initiativesAwaitingStatusUpdate: {
          ...state.initiativesAwaitingStatusUpdate,
          [action.payload.initiativeId]: false
        }
      };
  }
  return state;
};

const selectIdPayWallet = (state: GlobalState) => state.features.idPay.wallet;

export const idPayWalletInitiativeListSelector = createSelector(
  selectIdPayWallet,
  ({ initiatives }) =>
    pot.map(initiatives, ({ initiativeList }) => initiativeList)
);

export const idPayWalletSubscribedInitiativeListSelector = createSelector(
  idPayWalletInitiativeListSelector,
  initiativeListPot =>
    pot.map(initiativeListPot, list =>
      list.filter(({ status }) => status !== InitiativeStatus.UNSUBSCRIBED)
    )
);

export const idPayInitiativesFromInstrumentSelector = createSelector(
  selectIdPayWallet,
  ({ initiativesWithInstrument }) => initiativesWithInstrument
);

export const idPayWalletInitiativesListWithInstrumentSelector = createSelector(
  [idPayInitiativesFromInstrumentSelector],
  initiatives => pot.map(initiatives, ({ initiativeList }) => initiativeList)
);

export const idPayEnabledInitiativesFromInstrumentSelector = createSelector(
  [isIdPayEnabledSelector, idPayWalletInitiativesListWithInstrumentSelector],
  (isIdpayEnabled, initiatives) =>
    isIdpayEnabled ? pot.getOrElse(initiatives, []) : []
);

export const idPayAreInitiativesFromInstrumentLoadingSelector = createSelector(
  [isIdPayEnabledSelector, idPayInitiativesFromInstrumentSelector],
  (isIDPayEnabled, initiativesPot) =>
    isIDPayEnabled && pot.isLoading(initiativesPot)
);

export const idPayAreInitiativesFromInstrumentErrorSelector = createSelector(
  selectIdPayWallet,
  wallet => pot.isError(wallet.initiativesWithInstrument)
);

export const idPayInitiativesAwaitingUpdateSelector = createSelector(
  selectIdPayWallet,
  wallet => wallet.initiativesAwaitingStatusUpdate
);

export const idPayInitiativeFromInstrumentPotSelector = (
  initiativeId: string
) =>
  createSelector(
    [
      idPayEnabledInitiativesFromInstrumentSelector,
      idPayInitiativesAwaitingUpdateSelector
    ],
    (initiativesByInstrument, initiativesAwaitingUpdate) => {
      const initiative = initiativesByInstrument.find(
        i => i.initiativeId === initiativeId
      );
      const isItemActive =
        initiative?.status === InstrumentInitiativeStatus.ACTIVE;
      const isAwaitingUpdate = initiativesAwaitingUpdate[initiativeId];

      switch (isAwaitingUpdate) {
        case undefined:
          return pot.some(isItemActive);
        case true:
          return pot.someLoading(isItemActive);
        case false:
          return pot.someLoading(isItemActive);
        default:
          return pot.none;
      }
    }
  );

export default reducer;
