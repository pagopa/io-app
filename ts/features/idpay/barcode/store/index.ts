import * as pot from "@pagopa/ts-commons/lib/pot";
import * as O from "fp-ts/lib/Option";
import { pipe } from "fp-ts/lib/function";
import { createSelector } from "reselect";
import { getType } from "typesafe-actions";
import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
import { TransactionErrorDTO } from "../../../../../definitions/idpay/TransactionErrorDTO";
import { Action } from "../../../../store/actions/types";
import { GlobalState } from "../../../../store/reducers/types";
import { NetworkError } from "../../../../utils/errors";
import { calculateIdPayBarcodeSecondsToExpire } from "../utils";
import { idPayGenerateBarcode } from "./actions";

export type IdPayBarcodeState = {
  [initiativeId: string]: pot.Pot<
    TransactionBarCodeResponse,
    TransactionErrorDTO | NetworkError
  >;
};

const INITIAL_STATE: IdPayBarcodeState = {};

const reducer = (
  state: IdPayBarcodeState = INITIAL_STATE,
  action: Action
): IdPayBarcodeState => {
  switch (action.type) {
    case getType(idPayGenerateBarcode.request):
      return {
        ...state,
        [action.payload.initiativeId]: pipe(
          state[action.payload.initiativeId],
          O.fromNullable,
          O.fold(() => pot.noneLoading, pot.toLoading)
        )
      };
    case getType(idPayGenerateBarcode.success):
      return {
        ...state,
        [action.payload.initiativeId]: pot.some(action.payload)
      };
    case getType(idPayGenerateBarcode.failure):
      return {
        ...state,
        [action.payload.initiativeId]: pot.toError(
          state[action.payload.initiativeId],
          action.payload.error
        )
      };
  }
  return state;
};

// -------------------- SELECTORS ----------------------

const idPayBarcodeSelector = (state: GlobalState): IdPayBarcodeState =>
  state.features.idPay.barcode;

export const idPayBarcodeByInitiativeIdSelector = createSelector(
  idPayBarcodeSelector,
  state => (initiativeId: string) => state[initiativeId] ?? pot.none
  // type checker does not complain if the null check is removed, but
  // since it's a map access it could very well return undefined
);

// gracefully failing expire time selector
export const idPayBarcodeSecondsTillExpireSelector = createSelector(
  idPayBarcodeByInitiativeIdSelector,
  getInitiative => (initiativeId: string) =>
    pipe(
      pot.map(getInitiative(initiativeId), barcode =>
        calculateIdPayBarcodeSecondsToExpire(barcode)
      ),
      seconds => pot.getOrElse(seconds, 0)
    )
);

export default reducer;
