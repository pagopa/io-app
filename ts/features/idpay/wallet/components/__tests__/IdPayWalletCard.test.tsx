import * as React from "react";
import { createStore } from "redux";
import ROUTES from "../../../../../navigation/routes";
import { applicationChangeState } from "../../../../../store/actions/application";
import { appReducer } from "../../../../../store/reducers";
import { GlobalState } from "../../../../../store/reducers/types";
import { renderScreenWithNavigationStoreContext } from "../../../../../utils/testWrapper";
import { IdPayWalletCard } from "../IdPayWalletCard";

describe("IdPayWalletCard", () => {
  it("should match the snapshot", () => {
    const component = renderCard();
    expect(component).toMatchSnapshot();
  });
});

function renderCard() {
  const globalState = appReducer(undefined, applicationChangeState("active"));
  return renderScreenWithNavigationStoreContext<GlobalState>(
    () => (
      <IdPayWalletCard
        name="18 app"
        amount={9999}
        avatarSource={{
          uri: "https://vtlogo.com/wp-content/uploads/2021/08/18app-vector-logo.png"
        }}
        expireDate={new Date(2023, 11, 2)}
      />
    ),
    ROUTES.WALLET_HOME,
    {},
    createStore(appReducer, globalState as any)
  );
}
