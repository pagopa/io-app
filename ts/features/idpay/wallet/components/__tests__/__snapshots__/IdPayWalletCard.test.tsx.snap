// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdPayWalletCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              />
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 718,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  aria-level="1"
                  collapsable={false}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontSize": 17,
                      "fontWeight": "600",
                    }
                  }
                >
                  WALLET_HOME
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "#FFFFFF",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "aspectRatio": 1.6,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <RNSVGSvgView
                            bbHeight="100%"
                            bbWidth="100%"
                            focusable={false}
                            height="100%"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                Object {
                                  "flex": 0,
                                  "height": "100%",
                                  "width": "100%",
                                },
                              ]
                            }
                            width="100%"
                          >
                            <RNSVGGroup>
                              <RNSVGDefs>
                                <RNSVGClipPath
                                  name="clip"
                                >
                                  <RNSVGCircle
                                    cx="75%"
                                    cy="0"
                                    r={16}
                                  />
                                  <RNSVGCircle
                                    cx="75%"
                                    cy="100%"
                                    r={16}
                                  />
                                  <RNSVGRect
                                    height="100%"
                                    width="100%"
                                    x={0}
                                    y={0}
                                  />
                                </RNSVGClipPath>
                              </RNSVGDefs>
                              <RNSVGRect
                                clipPath="clip"
                                fill={4292733946}
                                height="100%"
                                propList={
                                  Array [
                                    "fill",
                                    "stroke",
                                    "strokeWidth",
                                    "strokeOpacity",
                                  ]
                                }
                                rx={16}
                                ry={16}
                                stroke={4278190080}
                                strokeOpacity={0.12}
                                strokeWidth={1}
                                width="100%"
                                x={0}
                                y={0}
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </View>
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "justifyContent": "space-between",
                              "paddingBottom": 16,
                              "paddingLeft": 16,
                              "paddingRight": 12,
                              "paddingTop": 12,
                            }
                          }
                        >
                          <View>
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "space-between",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                color="blueItalia-850"
                                defaultColor="black"
                                defaultWeight="SemiBold"
                                font="TitilliumWeb"
                                fontStyle={
                                  Object {
                                    "fontSize": 18,
                                    "lineHeight": 25,
                                  }
                                }
                                style={
                                  Array [
                                    Object {
                                      "fontSize": 18,
                                      "lineHeight": 25,
                                    },
                                    Object {
                                      "color": "#001F3D",
                                      "fontFamily": "Titillium Web",
                                      "fontStyle": "normal",
                                      "fontWeight": "600",
                                    },
                                  ]
                                }
                                weight="SemiBold"
                              >
                                18 app
                              </Text>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderColor": "rgba(14,15,19,0.1)",
                                      "borderCurve": "continuous",
                                      "borderWidth": 1,
                                      "overflow": "hidden",
                                    },
                                    Object {
                                      "backgroundColor": "#FFFFFF",
                                      "borderRadius": 8,
                                      "height": 44,
                                      "padding": 6,
                                      "width": 44,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#FFFFFF",
                                        "borderCurve": "continuous",
                                        "overflow": "hidden",
                                        "resizeMode": "contain",
                                      },
                                      Object {
                                        "borderRadius": 2,
                                      },
                                    ]
                                  }
                                >
                                  <Image
                                    onError={[Function]}
                                    source={
                                      Object {
                                        "uri": "https://vtlogo.com/wp-content/uploads/2021/08/18app-vector-logo.png",
                                      }
                                    }
                                    style={
                                      Object {
                                        "height": "100%",
                                        "width": "100%",
                                      }
                                    }
                                  />
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "height": 16,
                                }
                              }
                            />
                            <Text
                              allowFontScaling={false}
                              color="black"
                              defaultColor="black"
                              defaultWeight="SemiBold"
                              font="TitilliumWeb"
                              fontStyle={
                                Object {
                                  "fontSize": 16,
                                  "lineHeight": 21,
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "fontSize": 16,
                                    "lineHeight": 21,
                                  },
                                  Object {
                                    "color": "#0E0F13",
                                    "fontFamily": "Titillium Web",
                                    "fontStyle": "normal",
                                    "fontWeight": "600",
                                  },
                                ]
                              }
                              weight="SemiBold"
                            >
                              Disponibile
                            </Text>
                            <Text
                              allowFontScaling={false}
                              color="blueItalia-500"
                              defaultColor="bluegreyDark"
                              defaultWeight="SemiBold"
                              font="TitilliumWeb"
                              fontStyle={
                                Object {
                                  "fontSize": 24,
                                  "lineHeight": 34,
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "fontSize": 24,
                                    "lineHeight": 34,
                                  },
                                  Object {
                                    "color": "#0066CC",
                                    "fontFamily": "Titillium Web",
                                    "fontStyle": "normal",
                                    "fontWeight": "600",
                                  },
                                ]
                              }
                              weight="SemiBold"
                            >
                              9,999.00 €
                            </Text>
                          </View>
                          <Text
                            allowFontScaling={false}
                            color="blueItalia-850"
                            defaultColor="black"
                            defaultWeight="SemiBold"
                            font="TitilliumWeb"
                            fontStyle={
                              Object {
                                "fontSize": 16,
                                "lineHeight": 21,
                              }
                            }
                            style={
                              Array [
                                Object {
                                  "fontSize": 16,
                                  "lineHeight": 21,
                                },
                                Object {
                                  "color": "#001F3D",
                                  "fontFamily": "Titillium Web",
                                  "fontStyle": "normal",
                                  "fontWeight": "600",
                                },
                              ]
                            }
                            weight="SemiBold"
                          >
                            Valida fino al 12/23
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
